lib/components/OrderDialog/OrderDialog.tsx

```diff
@@ -95,64 +107,77 @@ export const OrderDialog: React.FC<OrderDialogProps> = ({
 
     setOrder(response.order)
     setOrderState(response.orderState)
-    setStage("order-progress")
+    setStage("progress")
+  }
+
+  const handleProgressContinue = async () => {
+    setLoading(true)
+    try {
+      // Simulate some async work
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+
+      setSteps((currentSteps) => {
+        const activeStepIndex = currentSteps.findIndex((step) => step.active)
+        if (activeStepIndex === -1) return currentSteps
+
+        const newSteps = [...currentSteps]
+        // Complete current step
+        newSteps[activeStepIndex] = {
+          ...newSteps[activeStepIndex],
+          completed: true,
+          active: false,
+        }
+
+        // If this was the last step, move to checkout
+        if (activeStepIndex === currentSteps.length - 1) {
+          setTimeout(() => setStage("checkout"), 500) // Small delay to show completion
+        } else {
+          // Activate next step
+          newSteps[activeStepIndex + 1] = {
+            ...newSteps[activeStepIndex + 1],
+            active: true,
+          }
+        }
+
+        return newSteps
```

Comment:
the Continue button isn't supposed to progress the steps, the steps should automatically progress. The best way to do this is like this:

1. Create a new fake endpoint `/_fake/orders/move_orders_forward`
2. Inside the _fixture_ (not the normal source code!!!!) call that endpoint on an interval, just poll it so that orders move forward naturally and change their state
3. In this code, use React Query to poll for the order (react query makes this really simple)